validation-test/compiler_crashers_fixed/28505-failed-call-arguments-did-not-match-up.swift

```diff
@@ -5,6 +5,6 @@
 // See https://swift.org/LICENSE.txt for license information
 // See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
 
-// RUN: not --crash %target-swift-frontend %s -emit-ir
+// RUN: not %target-swift-frontend %s -emit-ir
```

Comment:
I noticed this one passes with -swift-version 4 now that SE0110 is implemented. Independent of your fix we might want to loosen up the asserts or rethink the logic there for swift 3 mode.